### PR TITLE
cpp template package

### PR DESCRIPTION
**Pull request by AlexReimann**
Template package with pure cpp, no ros stuff

### CODE REVIEW

**Comment by asmodehn** [comment]
I see... so catkin + pure C++,
Do you plan to add the ROSC++ in this package, as an the executable calling the lib for example ?

**Comment by AlexReimann** [comment]
Yes, exactly. Splitting the library and the ROS part using it into two packages.

File: ros1_cpptemplate/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            COMPONENTS
+              roslint
+              rosconsole
+)
+
+##############################################################################
+# Catkin
+##############################################################################
+
+catkin_package(
```
Comment (asmodehn) [inline ros1_cpptemplate/CMakeLists.txt:22]: How about having the cpptemplate without catkin, using pure cmake stuff, to show pure c++ integration in ROS ?

File: ros1_cpptemplate/README.rst [hunk 2]
```diff
@@ -0,0 +1,39 @@
+ROS1 CPP Template
+====================
+
+This package is a template to be used as a reference when creating c++ libraries (and applications if wanted) to be used in a ROS package. The main part of the implementation should go into this package and be independent from ROS.
+The only packge dependencies are on the *roslint* for checking the coding style (build dependency only) and *rosconsole* to have a uniform way for logging. In this package we are following http://wiki.ros.org/BestPractices as much as possible, where it makes sense.
```
Comment (asmodehn) [inline ros1_cpptemplate/README.rst:5]: Can we use a non-ROS C++ linter ? maybe something embedded people are already familiar with?

Comment (asmodehn) [reply ros1_cpptemplate/README.rst:5]: Same for rosconsole: Any way to do output in a way that we can pluginto rosconsole later ( like apache's log4cxx ) ?

**Review by asmodehn — CHANGES_REQUESTED** [review]

File: ros1_cpptemplate/README.rst (hunk 2 above)
Comment (AlexReimann) [reply ros1_cpptemplate/README.rst:5]: What's the merit?
Sounds to me like ending up with two different systems.

**Comment by AlexReimann** [comment]
Hol, some pip stuff failed?

**Comment by asmodehn** [comment]
https://github.com/requests/requests/issues/3560

**Review by asmodehn — APPROVED** [review]

**Comment by asmodehn** [comment]
After discussion, lets keep the ROS tools in the C++ template, since C++ doesn't have "official" set of tools, nor guarantee cross-platform, like most recent languages. The ROS tools (although not my personal first choice) are as valid as any other.

**Pull request merged**